sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala

```diff
@@ -461,68 +462,269 @@ case class FindInSet(left: Expression, right: Expression) extends BinaryExpressi
+        return inputs(1).trim(inputs(0))
+      }
+    }
+    null
+  }
+
   override def doGenCode(ctx: CodegenContext, ev: ExprCode): ExprCode = {
-    defineCodeGen(ctx, ev, c => s"($c).trim()")
+    if (children.size == 2 && !children(0).isInstanceOf[Literal]) {
+      throw new AnalysisException(s"The trimming parameter should be Literal.")}
+
+    val evals = children.map(_.genCode(ctx))
+    val inputs = evals.map { eval =>
+      s"${eval.isNull} ? null : ${eval.value}"
+    }
+    val getTrimFunction = if (children.size == 1) {
+      s"UTF8String ${ev.value} = ${inputs(0)}.trim();"
+    } else {
+      s"UTF8String ${ev.value} = ${inputs(1)}.trim(${inputs(0)});".stripMargin
+    }
+    ev.copy(evals.map(_.code).mkString("\n") + s"""
+      boolean ${ev.isNull} = false;
+      ${getTrimFunction};
+      if (${ev.value} == null) {
+        ${ev.isNull} = true;
+      }
+    """)
+    }
+
+  override def sql: String = {
+    if (children.size == 1) {
+      val childrenSQL = children.map(_.sql).mkString(", ")
+      s"$prettyName($childrenSQL)"
+    } else {
+      val trimSQL = children(0).map(_.sql).mkString(", ")
+      val tarSQL = children(1).map(_.sql).mkString(", ")
+      s"$prettyName($trimSQL, $tarSQL)"
+    }
   }
 }
 
 /**
- * A function that trim the spaces from left end for given string.
+ * A function that trims the characters from left end for a given string, if the trimStr is not specified, it defaults
+ * to trim the spaces from the left end of the source string.
+ * trimStr: the function removes any characters from the left end of the source string which matches with the characters
+ * from trimStr, it stops at the first non-match character.
  */
 @ExpressionDescription(
-  usage = "_FUNC_(str) - Removes the leading and trailing space characters from `str`.",
+  usage = """
+    _FUNC_(str) - Removes the leading space characters from `str`.
+    _FUNC_(trimStr, str) - Removes the leading string contains the characters from the trim string from the `str`
+  """,
   extended = """
+    Arguments:
+      str - a string expression
+      trimStr - the trim string
     Examples:
-      > SELECT _FUNC_('    SparkSQL');
+      > SELECT _FUNC_('    SparkSQL   ');
        SparkSQL
+      > SELECT _FUNC_('Sp', 'SSparkSQLS');
+       arkSQLS
   """)
-case class StringTrimLeft(child: Expression)
-  extends UnaryExpression with String2StringExpression {
+case class StringTrimLeft(children: Seq[Expression])
+  extends Expression with ImplicitCastInputTypes {
 
-  def convert(v: UTF8String): UTF8String = v.trimLeft()
+  require (children.size <= 2 && children.nonEmpty,
+    "$prettyName requires at least one argument and no more than two.")
+
+  override def inputTypes: Seq[AbstractDataType] = Seq.fill(children.size)(StringType)
+  override def dataType: DataType = StringType
+
+  override def nullable: Boolean = children.exists(_.nullable)
+  override def foldable: Boolean = children.forall(_.foldable)
```

Comment:
Can we create a parent class for these classes? Since `inputeTypes, dataType, nullable, foldable` are all the same.